lib/components/sync-status/popover.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import classnames from 'classnames';
+import distanceInWordsToNow from 'date-fns/distance_in_words_to_now';
+import Popover from '@material-ui/core/Popover';
+
+import { getLastSyncedTime } from '../../utils/sync/last-synced-time';
+import getNoteTitles from './get-note-titles';
+
+class SyncStatusPopover extends React.Component {
+  render() {
+    const {
+      anchorEl,
+      classes = {},
+      id,
+      notes,
+      onClose,
+      theme,
+      unsyncedNoteIds,
+    } = this.props;
+    const themeClass = `theme-${theme}`;
+    const open = Boolean(anchorEl);
+    const hasUnsyncedChanges = unsyncedNoteIds.length > 0;
+
+    const QUERY_LIMIT = 10;
```

Comment:
Is this limit so that the popup doesn't become too tall? What if we put the note titles in a scrollable div?

Reply:
Good question. A few reasons for the limit, in my mind:

1. The note titles are taken by regex-ing the note contents on each sync indicator hover, and not cached.
1. I feel like having a scrollable area inside a hover-activated popover is a little too much complexity. That would warrant a proper click-activated popover.
1. The intent of this feature is not to provide an exhaustive list of unsynced changes, but rather a way to self-diagnose if something's going wrong. Providing a list of 100 note titles doesn't add much utility over 10 notes. If we really wanted to show every unsynced note in a useful way, we should consider a different UI.